### PR TITLE
Add User Metrics skeleton #116

### PR DESCRIPTION
**Pull request by alouanemed**
here is the final result

image

You can access this activity using the sign up button .

### CODE REVIEW

File: app/src/main/java/com/kanshu/kanshu/UserMetricsActivity.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        int id = item.getItemId();
+        return super.onOptionsItemSelected(item);
+    }
+
+
+    public class UserMetrcisPagerAdapter extends FragmentStatePagerAdapter {
+
+        private String[] pageTitles = {"My Prgoress", "My saved Characters", "Practice Exercices", "Add-ons"};
```
Comment (intrications) [inline app/src/main/java/com/kanshu/kanshu/UserMetricsActivity.java]: Correct spelling to Progress
Saved should start with a capital 's'
All strings should be extracted to strings.xml ready for translation.

File: app/src/main/java/com/kanshu/kanshu/UserMetricsActivity.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle action bar item clicks here. The action bar will
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        int id = item.getItemId();
+        return super.onOptionsItemSelected(item);
+    }
+
+
+    public class UserMetrcisPagerAdapter extends FragmentStatePagerAdapter {
```
Comment (intrications) [inline app/src/main/java/com/kanshu/kanshu/UserMetricsActivity.java]: Metrcis > Metrics

File: app/src/main/java/com/kanshu/kanshu/UserMyPrgoressFragment.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import android.view.View;
+import android.view.ViewGroup;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A placeholder fragment containing a simple view.
+ */
+public class UserMyPrgoressFragment extends Fragment {
```
Comment (intrications) [inline app/src/main/java/com/kanshu/kanshu/UserMyPrgoressFragment.java]: Spelling of Progress

File: app/src/main/java/com/kanshu/kanshu/UserPracticeExofragment.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+public class UserPracticeExofragment extends Fragment {
+
+    private LinearLayoutManager mLayoutManager;
+    /**
+     * The fragment argument representing the section number for this
+     * fragment.
+     */
+    private static final String ARG_SECTION_NUMBER = "section_number";
+
+    public static UserPracticeExofragment newInstance(int sectionNumber) {
```
Comment (intrications) [inline app/src/main/java/com/kanshu/kanshu/UserPracticeExofragment.java]: Fragment should be capitalised

File: app/src/main/res/layout/fragment_user_my_saved_chars.xml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context="com.kanshu.kanshu.ArticleActivity$PlaceholderFragment">
+
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:textAppearance="?android:attr/textAppearanceLarge"
+        android:text="My Saved Chars"
```
Comment (intrications) [inline app/src/main/res/layout/fragment_user_my_saved_chars.xml]: Extract to strings.xml - should be already there from the matching tab name.

**Comment by intrications** [comment]
Looks good.

Can you fix these issues?

Also, have you replaced the contents of ArticlesActivity? If so, can you create a new activity for this?

Then I will add a screen which links to all the Activities that are being created rather than each developer taking over the sign up button.

Thanks.

**Comment by intrications** [comment]
Also, it probably needs rebasing on master.

**Comment by alouanemed** [comment]
@intrications all fixed;
I updated ArticlesActivity's layout name which is used in UserMetricsActivity , they both have the same content.

that's weird I rebased before pushing this commits !

**Comment by intrications** [comment]
It's still got the old style navigation drawer that why I noticed that I don't think it was rebased.

File: app/src/main/java/com/kanshu/kanshu/ArticleActivity.java [hunk 6]
```diff
@@ -34,7 +32,7 @@
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_article);
+        setContentView(R.layout.activity_main);
```
Comment (intrications) [inline app/src/main/java/com/kanshu/kanshu/ArticleActivity.java]: Why was this layout name changed?

Comment (intrications) [inline app/src/main/java/com/kanshu/kanshu/ArticleActivity.java]: Oh, I see because they both have the same content.

Comment (intrications) [inline app/src/main/java/com/kanshu/kanshu/ArticleActivity.java]: main doesn't seem right for something that is reused. Can't think of a better name though. activity_with_nav_drawer_and_tabs seems too long.

Comment (alouanemed) [inline app/src/main/java/com/kanshu/kanshu/ArticleActivity.java]: main layout for now until we found something good !

Comment (intrications) [inline app/src/main/java/com/kanshu/kanshu/ArticleActivity.java]: Agreed.

**Comment by alouanemed** [comment]
cc @intrications  ;all good ?

**Comment by fernandobuzi** [comment]
@alouanemed @intrications Is it possible to have all the tab titles visible? Is it like that because the titles are too long?

Should we, maybe, try Progress/Saved Characters/Practice/Add-ons?

**Comment by fernandobuzi** [comment]
Unless the selected tab always aligns to the left? If that is so, I think it can stay as it is now.

**Comment by intrications** [comment]
@fernandobuzi There's no point trying to fit all the tabs in. Android phones go down to 3 inches and smaller. I'm just about to test the code but I suspect the tabs may left aligned like that.

**Comment by fernandobuzi** [comment]
;)

**Comment by intrications** [comment]
@alouanemed Have you checked your build works? Have you pushed all the files? I'm getting complaints about missing strings in strings.xml in UserMetricsActivity.

**Comment by intrications** [comment]
Also the actual UserPracticeExoFragment file does not appear to have been renamed. Only the declaration of class.

**Pull request closed without merging**